src/Popup.js

```diff
@@ -30,6 +30,48 @@ function calculateWithFallback (vp, lp, lc, kp, kc, vm, Δv) {
   }
 }
 
+/**
+ * getActualPosition -  positioning strategy only specifies the *preferred* strategy
+ *                     of the child popup. When the popup is rendered, it might
+ *                     reposition to the closest strategy that allows it to still
+ *                     be rendered on screen.
+ *                     This method returns the actual positioning used (out of all
+ *                     the strategies available)
+ *
+ * @param  {ClientRect} parentRect - a ClientRect with left,top,right,bottom,
+                                     width, and height properties for the parent
+ * @param  {Object} ChildRect - an object with left,top,width, height for child / popup
+ * @param {Object} gap - an object with x and y absolute gap values (positive numbers)
+ * @return String - a string that identifies the actual positioning strategy used
+ */
+export function getActualPosition(parentRect, childRect, gap) {
```

Comment:
I took a look and it's the best I've been able to come up far. `calculateWithFallback` has a lot of logic in which includes cryptic variable names. I'm not certain of what the k(p) and k(c) variables represent.